conseil-common/src/test/scala/tech/cryptonomic/conseil/common/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,106 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.concurrent.ExecutionContext
+
+import cats.effect._
+import org.http4s._
+import org.http4s.client.Client
+import fs2.Stream
+import io.circe.generic.auto._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.circe.CirceEntityEncoder._
+import org.scalatest.{Matchers, WordSpec}
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+class RpcClientTest extends WordSpec with Matchers {
+
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+  "Rpc Client" should {
+      "return a case class for valid json" in new HttpClientFixtures {
+        val response = """[{
+        |  "result": { "hash": "abc" },
+        |  "id": "requestId"
+        |}]""".stripMargin
+        val request = Stream(RpcRequest("2.0", "getBlock", Params(1), "requestId"))
+
+        getBlocks(request, response, 1).unsafeRunSync() shouldBe List(Block("abc"))
+      }
+
+      "allow to batch requests" in new HttpClientFixtures {
+        val response = """[
+        |  { "result": { "hash": "abc" }, "id": "1"},
+        |  { "result": { "hash": "def" }, "id": "2"},
+        |  { "result": { "hash": "ghj" }, "id": "3"}
+        |]""".stripMargin
+        val request = Stream(
+          RpcRequest("2.0", "getBlock", Params(1), "1"),
```

Comment:
What this `Params()` is used for, in context of these tests? I believe we can put whatever we want and this will still work?

Reply:
`Params` is defined in the same file. It's a mock for the dummy RPC method.